### ISSUE TITLE
reaperrr's Next+1 roadmap

### ISSUE THREAD

**Issue by reaperrr** [OpenRA/OpenRA#16769]
Mostly so I don't forget. Subject to change/getting more fleshed out.

- Tie up loose ends in Aircraft/aircraft activities
- More work towards docking refactor
- Refactor warheads (mostly `CreateEffectWarhead`)
- Get more AI improvements in

**Comment by pchote** [comment]
IMO we have more than enough good new features and improvements already merged or with open PRs that we should aim to converge on a new playtest series / prep branch ASAP after the hotfix is released.

My thoughts on the current milestone contents:
* ~PRs towards or regressions caused by my OpenGL/UI size rework. I have this under control, with all the remaining parts either finished or with fairly precise plans on how to finish/fix. IMO these are all essential for the next prep branch.~
* ~PRs towards unifying sprite handling (32bit support for world sprites, mod-defined formats for UI sprites). #16885 and #17489 cover most of the work towards this, but I don't have a good feeling for how much work the final unification of MappedImage to `ISpriteLoader` will take. These would be nice to have on the next prep branch, but it would be reasonable to bump these if it helps decrease reviewer workload fatigue and get the other work across the line.~
* ~GeoIP and IP sharing/privacy: Needs to be fixed, and i'll work on this at some point in the next couple of weeks.~
* ~Warhead refactoring: #17119 is an easy fix, and it would be a shame to miss it. #17014 is trickier, so might be best bumped to Next + 2 or later?~
* ~Pathfinding fixes: We should aim to wrap up any regressions or simple fixes~, but defer any more major improvements to Next + 2 or later - unless @teinarss can get the HPA implemention ready ASAP?
* AI improvements: I think these are important to the future of OpenRA, but I also think that we're all burnt out and probably can't deal with the AI code right now. Anything that isn't trivial should be bumped to Next + 2 or later.
* ~Rally point improvements: #17461 needs to be fixed. IMO it would be silly to not implement the rally point behaviour from https://github.com/OpenRA/OpenRA/pull/16701#issuecomment-554766034 to go with the rallypoint path queueing, but we should bump this if it ends up being one of the last milestone blockers.~
* ~Spectator army tab (#16709): I don't see any significant issue with completing this for the next prep branch. The next PR in my UI scaling implementation will improve the icon artwork scaling, which should help fix the icon readability. The missing icon crash shouldn't be too hard to fix.~
* ~Minelayer improvements: #17101 is straightforward enough.~ It would be nice if we could resurrect #17099 because it integrates nicely with the other minelayer improvements, and with the other input changes associated with free zooming - but we should bump this if it ends up being one of the last milestone blockers.

~Everything else on the milestone (#11614, #17067, #16682, etc) is probably best bumped to Next + 2 or later, and we should do a pass of the PR queue and possibly pull some others into the milestone.~

**Comment by reaperrr** [comment]
Yeah, other than wrapping up #17119 I don't really have anything that I consider important enough that it needs to be in next prep, although I'd still like some of my open PRs to be merged soonish, regardless of before or right after we split off next prep.

**Comment by reaperrr** [comment]
Considering we want this cycle to be short than last plus I don't know how much time and motivation I can bring up for the entries listed in the OP, I'm moving this to 'Future' for now. Closing it would reduce chances of me remembering too much, though.

**Comment by pchote** [comment]
>  maybe more work towards docking refactor

With ~~KKnD~~ OpenKrush back on the radar we could try again to work with @IceReaper on adapting and upstreaming https://github.com/IceReaper/OpenKrush/tree/master/OpenRA.Mods.OpenKrush/Mechanics/Docking
